### PR TITLE
Antdesign

### PR DESCRIPTION
**Pull request by Arun-42**
Adding Ant Design to the list of supported component libraries.

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@Arun-42 great work so far. Just FYI, we are preparing some breaking changes and we want to release v2 of this library soo. I suggest you take a look at the v2 branch. I would like to have this mapper released with the v2 changes.

Refactoring should not take that much and there are small changes to mappers themselves.

**Comment by rvsia** [comment]
@Hyperkid123 Don't worry, this branch is being built on V2.

**Comment by Hyperkid123** [comment]
HA. I can see now.

**Comment by Hyperkid123** [comment]
@Arun-42 can you please change the PR target to be the master branch, not v2 branch? We have officially released the V2. Thanks!

File: packages/ant-component-mapper/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+[![npm version](https://badge.fury.io/js/%40data-driven-forms%2Fmui-component-mapper.svg)](https://badge.fury.io/js/%40data-driven-forms%2Fmui-component-mapper)
+
+[![Data Driven Form logo](images/logo.png)](https://data-driven-forms.org/)
+
+Material-UI component mapper for [Data Driven Forms](https://github.com/data-driven-forms/react-forms).
```
Comment (rvsia) [inline packages/ant-component-mapper/README.md:5]: MUI mentions

File: packages/ant-component-mapper/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
+[![npm version](https://badge.fury.io/js/%40data-driven-forms%2Fmui-component-mapper.svg)](https://badge.fury.io/js/%40data-driven-forms%2Fmui-component-mapper)
```
Comment (rvsia) [inline packages/ant-component-mapper/README.md:1]: MUI mention

File: packages/ant-component-mapper/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+[![Data Driven Form logo](images/logo.png)](https://data-driven-forms.org/)
+
+Material-UI component mapper for [Data Driven Forms](https://github.com/data-driven-forms/react-forms).
+
+:book: For more information please visit the [documentation](https://data-driven-forms.org/). :book:
+
+Used by [ManageIQ](http://manageiq.org/), Red Hat Cloud Services.
```
Comment (rvsia) [inline packages/ant-component-mapper/README.md:9]: Please remove this line

File: packages/ant-component-mapper/demo/index.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+import FormRenderer from '@data-driven-forms/react-form-renderer';
+import 'antd/dist/antd.css';
+import './style.css';
+import demoSchema from '@data-driven-forms/common/src/demoschema';
+import { componentMapper, FormTemplate } from '../src';
+import wizardSchema from './demo-schemas/wizard-schema';
+
+
```
Comment (rvsia) [inline packages/ant-component-mapper/demo/index.js:11]: some empty line

File: packages/ant-component-mapper/package.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    "sass-loader": "^7.1.0",
+    "semantic-release": "15.12.0",
+    "style-loader": "^0.23.1",
+    "url-loader": "^1.1.2",
+    "webpack": "^4.25.1",
+    "webpack-cli": "^3.1.2",
+    "webpack-dev-server": "^3.1.10",
+    "webpack-merge": "^4.1.4"
+  },
+  "peerDependencies": {
```
Comment (rvsia) [inline packages/ant-component-mapper/package.json:55]: You have to put ant libraries here too, let users know that they have to install it separately

File: packages/ant-component-mapper/package.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    "@data-driven-forms/react-form-renderer": "^1.9.3",
+    "react": "^16.13.0",
+    "react-dom": "^16.13.0"
+  },
+  "release": {
```
Comment (rvsia) [inline packages/ant-component-mapper/package.json:60]: release can be removes, it's included in root package.json

File: packages/ant-component-mapper/rollup.config.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+const outputPaths = glob.sync(path.resolve(__dirname, './src/components/*.js'));
+
+const antExternals = createFilter(
+  [
+    'react',
+    'react-dom',
+    'prop-types',
+    '@data-driven-forms/react-form-renderer',
+    'antd/**'
```
Comment (rvsia) [inline packages/ant-component-mapper/rollup.config.js:23]: please inlude also:
```
'@data-driven-forms/react-form-renderer/**',
'antd'
```

File: packages/ant-component-mapper/src/common/form-wrapper.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Form } from 'antd';
+
+const AntForm = ({ label, children, layout, component, invalid, isRequired, help }) => {
+  return (
+    <Form layout={layout} component={component}>
```
Comment (rvsia) [inline packages/ant-component-mapper/src/common/form-wrapper.js:8]: You can try to remove the form here and put it in the FormWrapper in FormTemplate to make performance better

File: packages/ant-component-mapper/src/common/form-wrapper.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      <Form.Item
```
Comment (rvsia) [inline packages/ant-component-mapper/src/common/form-wrapper.js:9]: Note: We have to get rid of this somehow (I am afraid that we will have to mock this component), but in some future PR.

File: packages/ant-component-mapper/src/common/is-required.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { childrenPropTypes } from '@data-driven-forms/common/src/prop-types-templates';
+
+const styles = {
```
Comment (rvsia) [inline packages/ant-component-mapper/src/common/is-required.js:4]: Extract these styles to a file

File: packages/ant-component-mapper/scripts/update_demo.sh [hunk 11]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/bash
+
```
Comment (rvsia) [inline packages/ant-component-mapper/scripts/update_demo.sh:2]: Please remove this file

File: packages/ant-component-mapper/src/files/form-template.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { Button, Typography } from 'antd';
+import 'antd/dist/antd.css';
+import FormTemplate from '@data-driven-forms/common/src/form-template';
+
+const { Title, Paragraph } = Typography;
+
+const Form = ({ children, ...props }) => <form {...props}>{children}</form>;
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/form-template.js:9]: Here you can use the Form ant component

File: packages/ant-component-mapper/src/files/checkbox.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        value={input.name}
+      >
+        {isRequired ? <IsRequired>{label}</IsRequired> : label}
+      </AntCheckbox>
+    </AntForm>
+  );
+};
+
+SingleCheckbox.propTypes = {
+  input,
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/checkbox.js:34]: Input and meta are no provided as a prop, but they are obtained via useFieldApi hook, so you can remove them here (and also in all other components)

File: packages/ant-component-mapper/src/files/form-template.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { Button, Typography } from 'antd';
+import 'antd/dist/antd.css';
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/form-template.js:4]: Importing of styles should be handled by users in their application. You can remove it here.

File: packages/ant-component-mapper/src/files/sub-form.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+      <Row>
+        <Col span={24}>{renderForm(fields)}</Col>
+      </Row>
+    </Fragment>
+  );
+};
+
+SubForm.propTypes = {
+  fields: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
+  title: PropTypes.string,
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/sub-form.js:36]: Title, description, labels should be all `PropTypes.node` to allow put there anything.

Strings makes problems, when you are trying to use translations. For example, the default way for translating texts in React.Intl library is using `<FormattedMessage id={} defaultMessage={} />` and that is a element, so users would see proptypes errors.

File: packages/ant-component-mapper/src/files/tabs.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+import PropTypes from 'prop-types';
+import { useFormApi } from '@data-driven-forms/react-form-renderer';
+
+import { Tabs } from 'antd';
+
+const { TabPane } = Tabs;
+
+const FormTabs = ({ fields }) => {
+  const formOptions = useFormApi();
+  //const [activeTab, setActiveTab] = useState(0);
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/tabs.js:11]: comment here, same below

File: packages/ant-component-mapper/src/files/time-picker.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+const DatePicker = (props) => {
+  const { input, isReadOnly, isDisabled, placeholder, isRequired, label, helperText, description, validateOnMount, meta } = useFieldApi(props);
+  const invalid = validationError(meta, validateOnMount);
+  const placeholderDisplay = placeholder ? placeholder : 'Select time';
+  const help = invalid || helperText || description;
+  return (
+    <AntForm layout="vertical" isRequired={isRequired} label={label} invalid={invalid} help={help}>
+      <AntTimePicker
+        use12Hours
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/time-picker.js:17]: `format` and `use12Hours` should be passed via props (you can set their default values)

File: packages/ant-component-mapper/src/files/wizard/step-buttons.js [hunk 18]
```diff
@@ -0,0 +1,103 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+// import Button from '../button';
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/wizard/step-buttons.js:3]: Comments here + below

File: packages/ant-component-mapper/src/files/wizard/wizard-step.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import WizardStepButtons from './step-buttons';
+
+const style = {
+  minHeight: '250px'
```
Comment (rvsia) [inline packages/ant-component-mapper/src/files/wizard/wizard-step.js:6]: please extract this style into a file

File: packages/ant-component-mapper/package.json [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+    ]
+  },
+  "dependencies": {
+    "@date-io/date-fns": "^1.0.1",
+    "@date-io/moment": "^1.0.1",
+    "classnames": "^2.2.6",
+    "clsx": "^1.0.4",
+    "date-fns": "^1.30.1",
+    "moment": "^2.23.0",
+    "react-select": "^2.2.0"
```
Comment (rvsia) [inline packages/ant-component-mapper/package.json:79]: I don't see any of these deps being used here. If they are needed by the ant library, please include them in dev & peer deps.

**Comment by Hyperkid123** [comment]
@rvsia we can create a staging branch so we can merge it and finish the details over there.

**Comment by rvsia** [comment]
As we discussed privately: 

1) Change .css to .scss

2) update Switch to use 

```
checkedChildren={onText}
unCheckedChildren={offText}
```

3) the datepicker issue is caused by chrome autocompletion - no need to fix it

File: packages/ant-component-mapper/src/common/form-wrapper.js (hunk 8 above)
Comment (Arun-42) [reply packages/ant-component-mapper/src/common/form-wrapper.js:8]: The layout here decides if the label and item are horizontally or vertically aligned.  How can I pass the prop to FormWrapper from each component?

Comment (Arun-42) [reply packages/ant-component-mapper/src/common/form-wrapper.js:8]: Also on doing that, "required error" doesn't seem to work on Submit.

Comment (rvsia) [reply packages/ant-component-mapper/src/common/form-wrapper.js:8]: You can pass `layout` props to FormTemplate, so everyone can decide that layout the form will use and I think it should be common for all fields. (`vertical` can be default)

If requried error does not work, please check if there is no additional component between `Form` and `Form.Item`

**Comment by Hyperkid123** [comment]
@Arun-42 thanks for the PR. :pray:

**Pull request merged**